src/transport_ofi.h

```diff
@@ -658,6 +658,138 @@ void shmem_transport_put_nb(shmem_transport_ctx_t* ctx, void *target, const void
+                                    };
+        struct fi_msg_rma msg = {
+                                  .msg_iov = &msg_iov,
+                                  .desc = NULL,
+                                  .iov_count = 1,
+                                  .addr = GET_DEST(dst),
+                                  .rma_iov = &rma_iov,
+                                  .rma_iov_count = 1,
+                                  .context = frag_source,
+                                  .data = 0
+                                };
+
+        SHMEM_TRANSPORT_OFI_CTX_LOCK(ctx);
+        while (frag_source < (((uint8_t *) source) + len)) {
+            frag_len = MIN(shmem_transport_ofi_max_msg_size, 
+                          (size_t) (((uint8_t *) source) + len - frag_source));
+            polled = 0;
+
+            msg_iov.iov_base = frag_source;
+            msg_iov.iov_len = frag_len;
+
+            rma_iov.addr = frag_target;
+            rma_iov.len = frag_len;
+
+            msg.msg_iov = &msg_iov;
+            msg.rma_iov = &rma_iov;
+            msg.context = frag_source;
+
+            SHMEM_TRANSPORT_OFI_CNTR_INC(&ctx->pending_put_cntr);
+
+            do {
+                ret = fi_writemsg(ctx->ep, &msg, FI_DELIVERY_COMPLETE);
+            } while (try_again(ctx, ret, &polled));
+
+            frag_source += frag_len;
+            frag_target += frag_len;
+        }
+        SHMEM_TRANSPORT_OFI_CTX_UNLOCK(ctx);
+        if ((*completion) != -1) (*completion)++;
+    }
+
+    /* Transmit the signal */
+    shmem_transport_ofi_get_mr(sig_addr, pe, &addr, &key);
+    polled = 0;
+    ret = 0;
+
+    SHMEM_TRANSPORT_OFI_CTX_LOCK(ctx);
+    SHMEM_TRANSPORT_OFI_CNTR_INC(&ctx->pending_put_cntr);
+
+    const struct iovec msg_iov_signal = {
+                                          .iov_base = (uint8_t *) &signal,
+                                          .iov_len = sizeof(uint64_t)
+                                        };
+    const struct fi_rma_iov rma_iov_signal = {
+                                               .addr = (uint64_t) addr,
+                                               .len = sizeof(uint64_t),
+                                               .key = key
+                                             };
+    const struct fi_msg_rma msg_signal = {
+                                           .msg_iov = &msg_iov_signal,
+                                           .desc = NULL,
+                                           .iov_count = 1,
+                                           .addr = GET_DEST(dst),
+                                           .rma_iov = &rma_iov_signal,
+                                           .rma_iov_count = 1,
+                                           .context = (uint8_t *) &signal,
+                                           .data = 0
+                                         };
+
+    do {
+        ret = fi_writemsg(ctx->ep, &msg_signal, FI_DELIVERY_COMPLETE | FI_FENCE | FI_INJECT);
```

Comment:
@shefty @sayantansur Would this be a correct way to use FI flags for the signal transmission in put-with-signal (Ref: openshmem-org/specification#206)? 
Also, do we need to have `FI_DELIVERY_COMPLETE` here as we have already configured the endpoint with this flag during creation through 
```
tx_attr.op_flags          = FI_DELIVERY_COMPLETE;
```
In other words, do the flags provided with `fi_writemsg` gets OR'ed with the flags already configured? Or, they are replaced?

Reply:
What about `fi_atomicmsg`?  Manpage doesn't list `FI_DELIVERY_COMPLETE` as a flag on this operation?

Reply:
This was answered on #818.  Looks good.